packages/analytics-browser/src/storage/local-storage.ts

```diff
@@ -1,8 +1,21 @@
 import { getGlobalScope } from '@amplitude/analytics-client-common';
 import { BrowserStorage } from './browser-storage';
+import { Logger } from '@amplitude/analytics-types';
 
 export class LocalStorage<T> extends BrowserStorage<T> {
-  constructor() {
-    super(getGlobalScope()?.localStorage);
+  logger?: Logger;
+
+  constructor(logger?: Logger) {
+    super(getGlobalScope()?.localStorage, logger);
+  }
+
+  async set(key: string, value: T): Promise<void> {
+    if (Array.isArray(value) && value.length > 1000) {
+      const droppedEvents = value.splice(0, value.length - 1000);
```

Comment:
`splice` mutates the array and could have side effects when the same reference is used elsewhere. would suggest to use [pure functions](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976#:~:text=pure%20function%E2%80%9D%20mean%3F-,A%20pure%20function%20is%20a%20function,-which%3A)

this one is good article: https://medium.com/@fknussel/arrays-objects-and-mutations-6b23348b54aa

Reply:
Thanks Kevin for letting me know the trap. Use `slice()` instead